Library/ViewModels/LiveStreamContainerViewModel.swift

```diff
@@ -124,202 +123,291 @@ LiveStreamContainerViewModelInputs, LiveStreamContainerViewModelOutputs {
+          sessionConfig: OpenTokSessionConfig(
+            apiKey: $0.appId, sessionId: $0.sessionId, token: $0.token
+          )
         )
     }
 
-    let hideWhenReplay = Signal.merge(
-      self.viewDidLoadProperty.signal.mapConst(true),
-      event.map { !$0.liveNow }
-    ).skipRepeats()
+    let liveStreamType = Signal.merge(
+      Signal.combineLatest(hlsStreamUrl, useHlsStream.filter(isTrue)).map(first),
+      Signal.combineLatest(openTokSessionConfig, useHlsStream.filter(isFalse)).map(first)
+      )
+      .skipRepeats()
 
-    let hideWhenLive = Signal.merge(
+    let liveStreamEndedNormally = updatedEventFetch.values()
+      .map(didEndNormally(event:))
+
+    let observedGreenRoomOffOrInReplay = Signal.merge(
+      greenRoomOffStatus.filter(isTrue),
+      liveStreamEndedNormally.filter(isTrue)
+      )
+      .ignoreValues()
+
+    self.createVideoViewController = Signal.combineLatest(
+      liveStreamType,
+      observedGreenRoomOffOrInReplay
+      )
+      .map(first)
+
+    self.disableIdleTimer = Signal.merge(
       self.viewDidLoadProperty.signal.mapConst(true),
-      event.map { $0.liveNow }
-    ).skipRepeats()
+      self.viewDidDisappearProperty.signal.mapConst(false)
+    )
 
-    self.navBarTitleViewHidden = Signal.merge(
-      project.mapConst(true),
-      liveStreamControllerState.map { state in
-        switch state {
-        case .live(playbackState: .playing, _):   return false
-        case .replay(playbackState: .playing, _): return false
-        default:                                  return true
-        }
+    self.removeVideoViewController = self.createVideoViewController.take(first: 1)
+      .sample(on: greenRoomOffStatus.filter(isFalse).ignoreValues())
+      .ignoreValues()
+
+    let greenRoomState = greenRoomOffStatus
+      .filter(isFalse)
+
+    let replayState = liveStreamEndedNormally
+      .takePairWhen(self.videoPlaybackStateChangedProperty.signal.skipNil())
+      .filter { didEndNormally, playbackState in didEndNormally && !playbackState.isError }
+
+    let liveState = updatedEventFetch.values()
+      .takePairWhen(self.videoPlaybackStateChangedProperty.signal.skipNil())
+      .filter { event, playbackState in
+        event.liveNow && !playbackState.isError
       }
-    ).skipRepeats()
 
-    self.navBarLiveDotImageViewHidden = hideWhenReplay
-    self.creatorAvatarLiveDotImageViewHidden = hideWhenReplay
-    self.numberWatchingBadgeViewHidden = hideWhenReplay
-    self.availableForLabelHidden = Signal.combineLatest(nonStarter, hideWhenLive).map { $0 || $1 }
+    let nonStarterState = updatedEventFetch.values()
+      .map { event in isNonStarter(event: event) }
+      .filter(isTrue)
 
-    self.goToProject = configData
-      .takeWhen(self.goToProjectButtonPressedProperty.signal)
-      .map { project, liveStreamEvent, _, _ in
-        (project, liveStreamEvent.liveNow ? .liveStream : .liveStreamReplay)
+    let firebaseUserId = updatedEventFetch.values()
```

Comment:
So as we spoke about (moved here for discussion and removed the fixme). The code from line 302 - 324 does the following:

- Signs into Firebase to update the number of people watching value.
- If there is no token on the event object it signs in anonymously with an anonymous user ID
- If there is a token it authenticates as that user and returns their respective user ID.
- Observes the values from the producer to keep it alive as we don't do anything else with it.

Our discussion was how to test this and whether or not we want some kind of output for that. We aren't really too concerned if this errors out.